trieste/acquisition/rule.py

```diff
@@ -2073,6 +2073,191 @@ def update(
+                )
+            ),
+            1,
+        )[0]
+
+    def _init_eps(self) -> None:
+        global_lower = self.global_search_space.lower
+        global_upper = self.global_search_space.upper
+        self.eps = 0.5 * (global_upper - global_lower) / (5.0 ** (1.0 / global_lower.shape[-1]))
+
+    def _compute_global_distances(self) -> None:
+        # Pairwise distances along each axis in the global search space.
+        points = self.global_search_space.points
+        self._global_distances = tf.abs(
+            tf.expand_dims(points, -2) - tf.expand_dims(points, -3)
+        )  # [num_points, num_points, D]
+
+    def _update_neighbors(self) -> None:
+        # Indices of the neighbors within the trust region.
+        neighbors_mask = tf.reduce_all(
+            tf.gather(self._global_distances, self._location_ix) <= self.eps, axis=-1
+        )
+        neighbors_mask = tf.reshape(neighbors_mask, (-1,))
+        self._neighbor_ixs = tf.where(neighbors_mask)
+        self._neighbor_ixs = tf.squeeze(self._neighbor_ixs, axis=-1)
+        # Points within the trust region (including the location point).
+        self._points = tf.gather(self.global_search_space.points, self._neighbor_ixs)
+
+    def initialize(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> None:
+        """
+        Initialize the region by sampling a location from the global search space and selecting
+        the neighboring points within the trust region.
+
+        :param models: The model for each tag.
+        :param datasets: The dataset for each tag.
+        """
+
+        datasets = self.select_in_region(datasets)
+        self._init_location()
+        self._step_is_success = False
+        self._init_eps()
+        self._update_neighbors()
+        _, self._y_min = self.get_dataset_min(datasets)
+        self._initialized = True
+
+    def update(
+        self,
+        models: Optional[Mapping[Tag, ProbabilisticModelType]] = None,
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> None:
+        """
+        Update this region, including location and neighbors, using the given dataset. If the
+        size of the region is less than the minimum size, re-initialize the region.
+
+        If the new optimum improves over the previous optimum by some threshold (that scales
+        linearly with ``kappa``), the previous acquisition is considered successful.
+
+        If the previous acquisition was successful, the size is increased by a factor
+        ``1 / beta``. Conversely, if it was unsuccessful, the size is reduced by the factor
+        ``beta``.
+
+        :param models: The model for each tag.
+        :param datasets: The dataset for each tag.
+        """
+
+        if not self._initialized or tf.reduce_any(self.eps < self._min_eps):
+            self.initialize(models, datasets)
+            return
```

Comment:
Currently this isn't coordianated at all in the base implementation. Each sub-region only reacts and sees itself. So the whole product search space will not be initialised, only a part of it. This would need to be explicitly handled in a custom product class. However, it does seem like an important operation to support by default. I can look at how to propagate the intialisation up to the product... not sure how to do that yet. Do you think that is important to support here?

Reply:
@hstojic I have an idea, but it requires a new property and hence touching multiple classes. So I think it is best if I resolve this in a separate PR.

Reply:
Open PR #827 resolves this issue.

Reply:
ok, thats fair enough